cpp/src/arrow/table.h

```diff
@@ -286,7 +286,22 @@ struct ARROW_EXPORT ConcatenateTablesOptions {
   static ConcatenateTablesOptions Defaults() { return {}; }
 };
 
-/// \brief Construct table from multiple input tables.
+/// \brief Construct a new table from multiple input tables.
+///
+/// The new table is assembled from existing column chunks without copying,
+/// if schemas are identical. If schemas do not match exactly, an attempt is
+/// made to unify them, and then column chunks are converted to their
```

Comment:
@pitrou could docs of other functions be referenced from docstrings? E.g. it would make sense to link to docs of [PromoteTableToSchema](https://github.com/apache/arrow/blob/9d86ede69af14eacc7968684b7e9535cfc8e3907/cpp/src/arrow/table.h#L296-L310). Also `PromoteTableToSchema` docs doesn't seem to be public. Is that ideal?

Reply:
Yes, you can reference `PromoteTableToSchema` here. And, yes, we can publish the doc to the latter in the Sphinx API docs.

Reply:
Added.